src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    @Override
+    public void run() {
+        // Check if the trip check should be skipped (based on time, day of week, etc.)
+        if (shouldSkipMonitoredTripCheck(trip)) {
+            LOG.debug("Skipping check for trip: {}", trip.id);
+            return;
+        }
+        // Make a request to OTP with the monitored trip params or if this is in a test environment use the injected OTP
+        // response.
+        OtpDispatcherResponse otpDispatcherResponse = injectedOtpResponseForTesting != null
+            ? injectedOtpResponseForTesting
+            : OtpDispatcher.sendOtpPlanRequest(trip.queryParams);
+
+        if (otpDispatcherResponse.statusCode >= 400) {
+            // TODO: report bugsnag
+            LOG.error("Could not reach OTP server. status={}", otpDispatcherResponse.statusCode);
+            return;
+        }
+        Response otpResponse = otpDispatcherResponse.getResponse();
+        // TODO: Should null tripRequestId be fixed?
+        // TripSummary tripSummary = new TripSummary(otpDispatcherResponse.response.plan, otpDispatcherResponse.response.error, null);
+        // TODO: BIG - Actually find the specific itinerary to compare against. For now, just choose the first itin.
+        Itinerary itinerary = findComparisonItinerary(trip, otpResponse);
+        // BEGIN CHECKS
+        // Check for notifications related to service alerts.
+        checkTripForNewAlerts(trip, itinerary);
+        // Check for notifications related to delays
+        checkTripForDelays(trip, itinerary);
+        // Check for notifications related to itinerary changes
+        checkTripForItineraryChanges(trip, itinerary);
+        // Add latest OTP response.
+        trip.addResponse(otpResponse);
+        Persistence.monitoredTrips.replace(trip.id, trip);
+        // Send notification to user.
+        sendNotification();
+    }
+
+    private Itinerary findComparisonItinerary(MonitoredTrip trip, Response otpResponse) {
+        // TODO: BIG - Find the specific itinerary to compare against. For now, just choose the first itin.
+        return otpResponse.plan.itineraries.get(0);
+    }
+
+    private void checkTripForNewAlerts(MonitoredTrip trip, Itinerary itinerary) {
+        if (!trip.notifyOnAlert) {
+            LOG.debug("Notify on alert is disabled for trip {}. Skipping check.", trip.id);
+            return;
+        }
+        // FIXME: Need to somehow determine how to get last itinerary from responses in journey state.
+        Itinerary latestItinerary = trip.latestItinerary();
+        // Construct set from new alerts.
+        Set<LocalizedAlert> newAlerts = new HashSet<>(itinerary.getAlerts());
+        Set<LocalizedAlert> previousAlerts = new HashSet<>(latestItinerary.getAlerts());
+        // Unseen alerts consists of all new alerts that we did not previously track.
+        Set<LocalizedAlert> unseenAlerts = new HashSet<>(newAlerts);
+        unseenAlerts.addAll(previousAlerts);
```

Comment:
I think this should be a `removeAll` operation instead of `addAll`.

Reply:
Correct.